src/lib.rs

```diff
@@ -309,6 +318,64 @@ impl PyIterProtocol for PyEmbeddings {
     }
 }
 
+/// finalfusion vocab.
+#[pyclass(name=Vocab)]
+struct PyVocab {
+    embeddings: Rc<RefCell<EmbeddingsWrap>>,
+}
+
+#[pymethods]
+impl PyVocab {
+    fn item_to_indices(&self, key: String) -> PyResult<PyObject> {
+        let embeds = self.embeddings.borrow();
+
+        use EmbeddingsWrap::*;
+        let indices = match &*embeds {
+            View(e) => e.vocab().idx(key.as_str()),
+            NonView(e) => e.vocab().idx(key.as_str()),
+        };
+        if let Some(indices) = indices {
+            let gil = pyo3::Python::acquire_gil();
+            match indices {
+                WordIndex::Word(idx) => Ok([idx].to_object(gil.python())),
+                WordIndex::Subword(indices) => Ok(indices.to_object(gil.python())),
+            }
+        } else {
+            Err(exceptions::KeyError::py_err("Unknown word or n-grams"))
+        }
+    }
+}
+
+#[pyproto]
+impl PySequenceProtocol for PyVocab {
+    fn __len__(&self) -> PyResult<usize> {
+        let embeds = self.embeddings.borrow();
+
+        use EmbeddingsWrap::*;
+        match &*embeds {
+            View(e) => Ok(e.vocab().words().len()),
+            NonView(e) => Ok(e.vocab().words().len()),
+        }
+    }
+
+    fn __getitem__(&self, idx: isize) -> PyResult<String> {
+        let embeds = self.embeddings.borrow();
+
+        use EmbeddingsWrap::*;
+
+        let v = match &*embeds {
+            View(e) => e.vocab(),
+            NonView(e) => e.vocab(),
+        };
+
+        if (v.words().len() as isize) <= idx || idx < 0 {
```

Comment:
Isn't `idx >= v.words.len() as isize` more readable? Also, `as` has high precedence, so the parentheses are not needed. Nowadays you could also replace the condition by:

`(0..v.word.len()).contains(&idx)`

But I didn't check which Rust version added this (when later than 1.31.0, we can't use it yet).

Reply:
> Isn't idx >= v.words.len() as isize more readable? 

Sure!

>Also, as has high precedence, so the parentheses are not needed. 

Fixed.

>But I didn't check which Rust version added this (when later than 1.31.0, we can't use it yet).

Pretty sure `Range::contains` was added in a very recent version. 1.35 or even 1.36.